sdk/include/aditof/frame_definitions.h

```diff
@@ -55,30 +68,31 @@ enum class FrameDataType {
  */
 struct FrameDetails {
     /**
-     * @brief The width of the frame.
+     * @brief The type of the frame. Can be one of the types provided by the
+     * camera.
      */
-    unsigned int width;
+    std::string type;
 
     /**
-     * @brief The height of the frame.
+     * @brief A frame can have multiple types of data. For example it could
+     * hold data about depth and/or data about IR.
      */
-    unsigned int height;
+    std::vector<FrameDataDetails> dataDetails;
 
     /**
-     * @brief The width of the actual full frame.
+     * @brief The width of the entire frame
      */
-    unsigned int fullDataWidth;
+    unsigned int width;
```

Comment:
Given that the sizes of the contained frames are specified in `dataDetails` I think it would be more useful to have just `size` in `FrameDetails`. This would make more sense in the case we include metadata in the frame (which is not organized in a matrix fashion)